### PR TITLE
Add recombination and mismatch params to cli

### LINKED ISSUE #435: Add mismatch and recombination_rate arguments to the command line
The CLI currently does not allow control of these parameters.

### PR DESCRIPTION
**Pull request by hyanwong**
Fixes #435

### CODE REVIEW

**Review by jeromekelleher — COMMENTED** [review]
Very nice, clean and easy. 

I guess we need a dependency on msprime now, which we should raise as an issue to make sure we track the dependencies in (e.g.) the conda package.

**Comment by hyanwong** [comment]
> Very nice, clean and easy.
> 
> I guess we need a dependency on msprime now, which we should raise as an issue to make sure we track the dependencies in (e.g.) the conda package.

It's already in CI-tests-complete and CI-tests-conda. Not in CI-tests-pip though, but presumably is pulled in automatically when installing from pip?

**Comment by jeromekelleher** [comment]
> It's already in CI-tests-complete and CI-tests-conda. Not in CI-tests-pip though, but presumably is pulled in automatically when installing from pip?

This is for the actual packaging, not just tests. Need to be much more careful about adding run-time dependencies.

We just need to track it as an issue, no need to get into it here.

**Comment by hyanwong** [comment]
https://github.com/tskit-dev/tsinfer/issues/736

**Comment by hyanwong** [comment]
I've no idea how I can test this, because calling the CLI command in the test suite doesn't end up the the mismatch etc parameters dumped into the tree sequence. So I still have no way to check from the ts file generated within the test suite that the mismatch etc parameters were passed correctly.

**Comment by jeromekelleher** [comment]
Mocking out some key functions is the right way I think, as this will tell you whether values were passed through in the expected way.

**Comment by hyanwong** [comment]
OK, I used mocking. Thanks for the tip. I think this should be ready to merge

**Comment by hyanwong** [comment]
Also fixes https://github.com/tskit-dev/tsinfer/issues/736

File: tests/test_cli.py [hunk 1]
```diff
@@ -329,6 +329,106 @@ def test_chain(self):
...
+        command = [
+            "infer",
+            self.sample_file,
+            "--recombination_map",
+            ratemap,
+        ]
+        with mock.patch("tsinfer.infer") as infer:
+            self.run_command(command)
+            args, kwargs = infer.call_args
+            assert type(kwargs["recombination_rate"]) == msprime.RateMap
```
Comment (jeromekelleher) [inline tests/test_cli.py:405]:
```suggestion
            assert isinstance(kwargs["recombination_rate"]), msprime.RateMap)
```

**Review by jeromekelleher — APPROVED** [review]
LGTM, one minor tweak.

File: tests/test_cli.py (hunk 1 above)
Comment (hyanwong) [reply tests/test_cli.py:405]: Ah yes, much better, thanks.

**Pull request merged**